nasa/harmony
# HARMONY_1702: As a harmony operator, I want to ensure harmony does not need to be restarted when a backend service version is updated

## Jira Issue ID
HARMONY_1702

## Description
Add `deployment-callback` endpoint to allow updating harmony server with the new service image. This endpoint is supposed to be called only by the ci-cd `bin/deploy-service` script as part of the service self-deployment workflow.

## Local Test Steps
Tested in sandbox https://internal-harmony-yliu10-frontend-1209292950.us-west-2.elb.amazonaws.com,
Pick a harmony service to test, e.g. giovanni-adapter,
1. Submit a giovanni request, note down the service image tag of the successful request in workflow ui. 
https://internal-harmony-yliu10-frontend-1209292950.us-west-2.elb.amazonaws.com/C1225808238-GES_DISC/ogc-api-coverages/1.0.0/collections/Grid%2FprecipitationCal/coverage/rangeset?format=text%2Fcsv&point=0.76,-3.8&subset=time(%222020-01-06T12%3A00%3A00Z%22%3A%222020-01-06T16%3A00%3A00Z%22)

2. Call {{ _.root }}/service-image-tag/giovanni-adapter endpoint to update the giovanni-adapter to a different image with request body: `{"tag": "yliu10"}`

3. Wait one minute for the service self-deployment to run to completion. Then submit another giovanni request, verify that the request runs successfully and it is processed by the new giovanni service image with tag `yliu10`.

## PR Acceptance Checklist
